sssd_test_framework/hosts/keycloak.py

```diff
@@ -46,6 +46,28 @@ def __init__(self, *args, **kwargs) -> None:
 
         self.adminpw = self.config.get("adminpw", "Secret123")
 
+        self.client.setdefault("id_provider", "idp")
+        self.client.setdefault("enabled", "true")
+        self.client.setdefault("idp_type", "keycloak:https://master.keycloak.test:8443/auth/admin/realms/master/")
+        self.client.setdefault("auto_private_groups", "true")
+        self.client.setdefault("use_fully_qualified_names", "true")
+        self.client.setdefault("debug_level", "9")
+        self.client.setdefault("idp_client_id", "myclient")
+        self.client.setdefault("idp_client_secret", "ClientSecret123")
+        self.client.setdefault(
+            "idp_token_endpoint", "https://master.keycloak.test:8443/auth/realms/master/protocol/openid-connect/token"
+        )
+        self.client.setdefault(
+            "idp_userinfo_endpoint",
+            "https://master.keycloak.test:8443/auth/realms/master/protocol/openid-connect/userinfo",
+        )
+        self.client.setdefault(
+            "idp_device_auth_endpoint",
+            "https://master.keycloak.test:8443/auth/realms/master/protocol/openid-connect/auth/device",
+        )
+        self.client.setdefault("idp_id_scope", "profile")
+        self.client.setdefault("idp_auth_scope", "openid profile email")
+
```

Comment:
Doesn't this also need `self.domain = self.client["idp_domain"]` as in 
https://github.com/SSSD/sssd-test-framework/blob/3db73377684e3a697bfc932b1d4b7295c9a33024/sssd_test_framework/hosts/ipa.py#L66
?

Reply:
Hi,

if I understand it correctly this is not needed if the client should just use the default domain name `test`. Since there is not strict requirement on the domain name in the IdP provider I think we can just use `test`.

bye,
Sumit

Reply:
But this might result in multiple domains named 'test' if some other domain doesn't define non-default name, no?

Reply:
Hi,

if I understand it correctly, if you want to change the name if the SSSD domain you have to change the name in the topology definition, e.g.
```
--- a/sssd_test_framework/topology.py
+++ b/sssd_test_framework/topology.py
@@ -123,7 +123,7 @@ class KnownTopology(KnownTopologyBase):
         name="keycloak",
         topology=Topology(TopologyDomain("sssd", client=1, keycloak=1)),
         controller=KeycloakTopologyController(),
-        domains=dict(test="sssd.keycloak[0]"),
+        domains=dict(keycloak="sssd.keycloak[0]"),
         fixtures=dict(
             client="sssd.client[0]",
             keycloak="sssd.keycloak[0]",
```
But I'm not sure if this is expected here. @pbrezina, do you have any recommendations?

bye,
Sumit

Reply:
The domain Alexey is referring to is the IPA domain used to realm join, not SSSD domain. Maybe it does not make sense here? Or maybe we can set it to "keycloak.test"?